storage/innobase/handler/ha_innodb.cc

```diff
@@ -11922,21 +11922,20 @@ void ha_innobase::adjust_encryption_options(HA_CREATE_INFO *create_info,
 
   bool is_tmp = (create_info->options & HA_LEX_CREATE_TMP_TABLE) != 0;
 
-  /* If table is intrinsic, it will use encryption for table based on
-  temporary tablespace encryption property. For non-intrinsic tables
-  without explicit encryption attribute, table will be forced to be
-  encrypted if innodb_encrypt_tables=ON/FORCE */
+  if (is_intrinsic || is_tmp) {
+	  return;
+  }
+
   if (create_info->encrypt_type.length == 0 &&
       create_info->encrypt_type.str == nullptr) {
-    if ((is_intrinsic && srv_tmp_space.is_encrypted()) ||
-        (!is_intrinsic && (srv_encrypt_tables == SRV_ENCRYPT_TABLES_ON ||
-                           srv_encrypt_tables == SRV_ENCRYPT_TABLES_FORCE))) {
+    if (srv_encrypt_tables == SRV_ENCRYPT_TABLES_ON ||
+        srv_encrypt_tables == SRV_ENCRYPT_TABLES_FORCE) {
       create_info->encrypt_type = yes_string;
-    } else if (!is_intrinsic && !is_tmp &&
-               (srv_encrypt_tables == SRV_ENCRYPT_TABLES_KEYRING_ON ||
-                srv_encrypt_tables == SRV_ENCRYPT_TABLES_KEYRING_FORCE ||
-                srv_encrypt_tables ==
-                    SRV_ENCRYPT_TABLES_ONLINE_TO_KEYRING_FORCE)) {
+    }
+
+    if (srv_encrypt_tables == SRV_ENCRYPT_TABLES_KEYRING_ON ||
```

Comment:
Later: should be else if? But consider converting to switch (srv_encrypt_tables) instead